lib/script/script.js

```diff
@@ -713,14 +716,19 @@ Script.fromAddress = function(address) {
 
 /**
  * @param {Network=} network
- * @return {Address} the associated address for this script
+ * @return {Address|boolean} the associated address for this script if possible, or false
  */
 Script.prototype.toAddress = function(network) {
+  var Address = require('../address');
   network = Networks.get(network) || this._network || Networks.defaultNetwork;
-  if (this.isPublicKeyHashOut() || this.isScriptHashOut()) {
+  var canConvertToAddress = this.isPublicKeyHashOut() ||
+    this.isScriptHashOut() ||
+    this.isPublicKeyHashIn() ||
+    this.isScriptHashIn();
+  if (canConvertToAddress) {
     return new Address(this, network);
   }
-  throw new Error('The script type needs to be PayToPublicKeyHash or PayToScriptHash');
+  return false;
```

Comment:
Returning a boolean sounds better, that way checking isScriptHashOut, etc isn't needed prior, and checking then checking for false instead. May be good to bump to v0.12.0.